statsmodels/tools/validation/validation.py

```diff
@@ -395,6 +371,27 @@ def float_like(value, name, optional=False, strict=False):
 
 
 def string_like(value, name, optional=False, options=None, lower=True):
+    """
+    Check if of type string and raise not string_like if not
```

Comment:
This doesn't make much sense.  Maybe just "Check if object is string-like and raise if not"